nuagekubemon/policy/resourcemanager.go

```diff
@@ -136,8 +137,70 @@ func (rm *ResourceManager) GetPolicyGroupsForPod(podName string, podNs string) (
 	return &pgList, nil
 }
 
+const (
+	zoneAnnotationAdd    = "ZoneAnnotationAdd"
+	zoneAnnotationRemove = "ZoneAnnotationRemove"
+)
+
+func (rm *ResourceManager) createNuagePolicyUpdateForZoneAnnotation(namespace string, op string) error {
+
+	enterprise, ok := rm.vsdMeta["enterpriseName"]
+	if !ok {
+		glog.Error("Failed to get enterprise for namespace annotation operation")
+		return errors.New("Failed to get enterprise for namespace annotation operation")
+	}
+
+	domain, ok := rm.vsdMeta["domainName"]
+	if !ok {
+		glog.Error("Failed to get domain for namespace annotation operation")
+		return errors.New("Failed to get domain for namespace annotation operation")
+	}
+
+	nuagePolicy := policies.NuagePolicy{
+		Version:    policies.V1Alpha,
+		Type:       policies.Default,
+		Enterprise: enterprise,
+		Domain:     domain,
+		Name:       api.ZoneAnnotationTemplateName,
+	}
+
+	defaultPolicyElement := policies.DefaultPolicyElement{
+		Name:   fmt.Sprint("Namespace annotation for %s", namespace),
+		From:   policies.EndPoint{Name: namespace, Type: policies.Zone},
+		To:     policies.EndPoint{Name: namespace, Type: policies.EndPointZone},
+		Action: policies.Deny,
+		NetworkParameters: policies.NetworkParameters{
+			Protocol:             policies.TCP,
```

Comment:
Its number and has to be either tcp or udp

https://github.com/nuagenetworks/vsd-api-specifications/blob/master/ingressaclentrytemplate.spec#L932